### DISCUSSION TITLE
How to use setFPSCamera method of GameScene

### DISCUSSION THREAD

**Discussion by feiniao5643** [AlmasB/FXGL#1185] [General]
First of all, please forgive me for my bad English.
I'll skip the usual preliminaries and come straight to the point.
I use this API in initGame,It's normal.
`@Override
    protected void initGame() {
        camera3D = FXGL.getGameScene().getCamera3D();
        FXGL.getGameWorld().addEntityFactory(new GameEntityFactory());
        camera3D.setMoveSpeed(500);
        camera3D.getTransform().setZ(-200);
        camera3D.getTransform().setAngle(30);
        **FXGL.getGameScene().setFPSCamera(true);**
        FXGL.spawn("铁骑兵",0,0,0);
    }`
But I can't use it in mouse events.
`FXGL.getInput().addAction(new UserAction("Mouse Turn") {
            @Override
            protected void onActionBegin() {
                FXGL.getGameScene().setFPSCamera(true);
            }

            @Override
            protected void onActionEnd() {
                System.out.println(0);
                FXGL.getGameScene().setFPSCamera(false);
            }
        },MouseButton.SECONDARY);`
Finally, the function I want to achieve is that when the right mouse button is pressed, the camera moves with the mouse, and the camera is fixed after release.

**Comment by AlmasB** [comment]
Thanks, I reproduced this and will try to figure out why.

**Comment by AlmasB** [comment]
Right, the issue was with the mouse buttons being pressed (which fired MOUSE_DRAGGED rather than MOUSE_MOVED). It is now fixed, thanks for raising this. Your code above should now work fine.

To get the latest build, you can use this:
```
<dependencies>
    <dependency>
        <groupId>com.github.almasb</groupId>
        <artifactId>fxgl</artifactId>
        <version>17+dev-SNAPSHOT</version>
    </dependency>
</dependencies>

<repositories>
    <repository>
        <id>oss.sonatype.org-snapshot</id>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>
```
